core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnTimezoneFromTime.java

```diff
@@ -56,8 +57,22 @@ private static ISequence<IDayTimeDurationItem> execute(@NonNull IFunction functi
       IItem focus) {
     ITimeItem arg = FunctionUtils.asTypeOrNull(ObjectUtils.requireNonNull(arguments.get(0).getFirstItem(true)));
 
-    return arg == null || !arg.hasTimezone()
+    return arg == null
         ? ISequence.empty()
-        : ISequence.of(arg.getOffset());
+        : ISequence.of(fnTimezoneFromDate(arg));
+  }
+
+  /**
+   * Implements <a href=
+   * "https://www.w3.org/TR/xpath-functions-31/#func-timezone-from-time">fn:timezone-from-time</a>.
+   *
+   * @param arg
+   *          the meta:time item from which to extract the timezone component
+   * @return the timezone component from the date/time or {@code null} if no
+   *         timezone is present
+   */
+  @Nullable
+  public static IDayTimeDurationItem fnTimezoneFromDate(@NonNull ITimeItem arg) {
```

Comment:
Shouldn't the util function name match the class?

```suggestion
  public static IDayTimeDurationItem fnTimezoneFromTime(@NonNull ITimeItem arg) {
```

Reply:
fixed this in the latest commit, since I had to update the method call as well.